File: kotlin/courses/jungdaeri-kotlin-basics/function/Function.kt
fun main() {

    fun sayHello() {
        println("Hello")
    }

    fun sayHelloWithName(name: String) {
        println("Hello, $name")
    }

    fun sayHelloWithDefaultName(name: String = "Kim") {
        println("Hello, $name")
    }

    fun sayHelloWithReturns(): String {
        return "Hello"
    }

    sayHello()
    sayHelloWithName("Test")
    sayHelloWithDefaultName()
    println(sayHelloWithReturns())
}

File: languages/dart/courses/jungdaeri-dart-basics/function.dart
void main() {
  saySomething();

  String hello = sayHello('kim');

  print(hello);
  sayMessage('Hello, world!', isDone: false);

  void Function() greeting = () {
    print('Hello');
  };

  greeting();
}

void saySomething() {
  print('Something');
}

// String sayHello(String name) {
//   return 'Hello, $name';
// }

String sayHello(String name) => 'Hello, $name';

void sayMessage(String message, {bool isDone = true}) {
  print('$message / isDone: $isDone');
}
